dtreeviz/models/shadow_decision_tree.py

```diff
@@ -438,17 +444,26 @@ def _get_y_data(y_data):
 
     @staticmethod
     def get_shadow_tree(tree_model, x_data, y_data, feature_names, target_name, class_names=None, tree_index=None):
+        if hasattr(tree_model, 'get_booster'):
+            tree_model = tree_model.get_booster() # support XGBClassifier and XGBRegressor
         if isinstance(tree_model, ShadowDecTree):
             return tree_model
         elif isinstance(tree_model, (sklearn.tree.DecisionTreeRegressor, sklearn.tree.DecisionTreeClassifier)):
             from dtreeviz.models import sklearn_decision_trees
             return sklearn_decision_trees.ShadowSKDTree(tree_model, x_data, y_data, feature_names,
                                                         target_name, class_names)
-        elif isinstance(tree_model, xgboost.core.Booster):
+        elif str(type(tree_model)).endswith("xgboost.core.Booster'>"):
```

Comment:
I'd have to check this myself but this seems like a complicated expression that we could simplify

Reply:
Could do something like [safe_isinstance()](https://github.com/slundberg/shap/blob/8a45ff574fb0cad4c2d3cee8e5a825c7ae5f82d8/shap/utils/_general.py#L172) like in the shap library for a slightly more elegant way of checking instance type without actually loading a module.

Reply:
since str(type(tree_model)) is used in many places, we could create from it a variable named tree_model_type, for example, and use it like `elif 'xgboost.core.Booster' in tree_model_type:`

Reply:
Probably okay just to reference the `__name__` via its type.